edx_proctoring/static/proctoring/js/views/proctored_exam_onboarding_view.js

```diff
@@ -0,0 +1,239 @@
+        getOnboardingStatus: function(status) {
+            if (onboardingStatuses.includes(status)) {
+                return onboardingStatusReadableFormat[status];
+            } else {
+                return status;
+            }
+        }
+    };
+    edx.instructor_dashboard.proctoring.ProctoredExamOnboardingView = Backbone.View.extend({
+        initialize: function() {
+            this.setElement($('.student-onboarding-status-container'));
+            this.collection = new edx.instructor_dashboard.proctoring.ProctoredExamOnboardingCollection();
+            this.templateUrl = '/static/proctoring/templates/student-onboarding-status.underscore';
+            this.courseId = this.$el.data('course-id');
+            this.template = null;
+
+            this.initialUrl = this.collection.url;
+            this.collection.url = this.initialUrl + this.courseId;
+            this.inSearchMode = false;
+            this.searchText = '';
+            this.filters = [];
+            this.currentPage = 1;
+
+            /* re-render if the model changes */
+            this.listenTo(this.collection, 'change', this.collectionChanged);
+
+            /* Load the static template for rendering. */
+            this.loadTemplateData();
+        },
+        events: {
+            'click .search-onboarding > span.search': 'searchItems',
+            'click .search-onboarding > span.clear-search': 'clearSearch',
+            'click .status-checkboxes > li > input': 'filter',
+            'click li > a.target-link': 'getPaginatedAttempts'
+        },
+        searchItems: function(event) {
+            var searchText = $('#search_onboarding_id').val();
+            if (searchText !== '') {
+                this.inSearchMode = true;
+                this.searchText = searchText;
+                this.currentPage = 1;
+                this.collection.url = this.constructUrl();
+                this.hydrate();
+                event.stopPropagation();
+                event.preventDefault();
+            }
+        },
+        clearSearch: function(event) {
+            this.inSearchMode = false;
+            this.searchText = '';
+            this.currentPage = 1;
+            this.collection.url = this.constructUrl();
+            this.hydrate();
+            event.stopPropagation();
+            event.preventDefault();
+        },
+        filter: function(event) {
+            var $checkbox = $(event.currentTarget);
+            var checked = $checkbox.is(':checked');
+            var value = $checkbox.val();
+            var index = this.filters.indexOf(value);
+            this.currentPage = 1;
+            // check the value to prevent adding it twice from double clicking
+            if (checked && !this.filters.includes(value)) {
+                this.filters.push(value);
+            } else {
+                this.filters.splice(index, 1);
+            }
+            this.collection.url = this.constructUrl();
+            this.hydrate();
```

Comment:
If possible it would be nice to have a small debounce on making this backend call so when a user clicks through multiple boxes we don't call the backend necessarily a bunch of times.

Reply:
I went ahead and added submit/clear buttons, so it will only make the call on submit rather than with each checkbox clicked.